transcom/mymove
# INT B-19940 Diversion Request Modal

## [B-19940](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A948733)

## Summary

Adds a modal between pressing "request diversion" and the request being made. Requires the user/TOO to provide a reason for their requested diversion.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the office app
2. Login as a TOO
3. Find a move with an HHG shipment that has been approved by the TOO and is not yet diverted, requested to be diverted, cancelled, or requested to be cancelled. #BILWEI should work if it is fresh.
4. Navigate to the move task order page.
5. Locate and click the Request Diversion button.
6. A modal should open.
7. Ensure you are unable to submit the modal without a providing a reason.
8. When you submit with a reason ensure, in the dev tools, that the request and response include your reason.
9. To create an error, attempt to submit another diversion request for the same shipment. A flash message should display saying there was an error should appear. The ability to re-request on an already requested or diverted shipment will be prevented in an upcoming backlog item.


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
<img width="644" alt="Screenshot 2024-05-23 at 9 59 06 AM" src="https://github.com/transcom/mymove/assets/158500142/8f29bfdb-e654-4851-b85b-7eac96faaf94">
